src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,64 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+
+/**
+ * Represents a Storage object that handles the loading and saving of tasks.
+ */
+public class Storage {
+    private String filePath;
+    private File file;
+    private Scanner sc;
+
+    /**
+     * Creates a Storage object.
+     *
+     * @param filePath The path to the file where the tasks are stored.
+     * @throws IOException If the file does not exist and cannot be created.
+     */
+    public Storage(String filePath) throws IOException {
+        this.filePath = filePath;
+        this.file = new File(filePath);
+        if (!file.exists()) {
+            System.out.println("File does not exist. Creating file...");
```

Comment:
Perhaps the print statements from here can be moved to Ui class for more encapsulation?

Reply:
I agree. However, it was a bit hard to abstract them properly.